File: packages/sdk/android/core/src/main/java/io/twohire/bleintsdk/bluetooth/BluetoothError.kt
package io.twohire.bleintsdk.bluetooth

import io.twohire.bleintsdk.utils.ErrorDescription

internal enum class BluetoothError : ErrorDescription {
    NOT_CONNECTED {
        override val description: String
            get() = "Bluetooth is not in connected state"
    },
    PERIPHERAL_NOT_FOUND {
        override val description: String
            get() = "Bluetooth peripheral not found"
    },
    NOT_READING {
        override val description: String
            get() = "Bluetooth is not reading"
    },
    TIMEOUT_ERROR {
        override val description: String
            get() = "Bluetooth timeout error"
    },
    ADAPTER_NOT_ENABLED {
        override val description: String
            get() = "Bluetooth adapter is not enabled"
    },
    GENERIC {
        override val description: String
            get() = "Bluetooth generic error"
    }
}

File: packages/sdk/ios/core/BLEIntSDK/Classes/Bluetooth/BluetoothError.swift
//
//  BluetoothError
//
//  Copyright © 2022 2hire.io. All rights reserved.
//

import Foundation

internal enum BluetoothError: Error {
    case Generic
    case NotConnected
    case NotReading
    case PeripheralNotFound
    case CharacteristicNotFound
    case ApiMisuse
    case Timeout
}

extension BluetoothError: LocalizedError {
    public var errorDescription: String? {
        switch self {
        case .ApiMisuse:
            return NSLocalizedString("Bluetooth Api misuse, an action is already running", comment: "")
        case .NotConnected:
            return NSLocalizedString("Bluetooth is not in connected state", comment: "")
        case .NotReading:
            return NSLocalizedString("Bluetooth is not in Reading state", comment: "")
        case .Timeout:
            return NSLocalizedString("Bluetooth action timeout error", comment: "")
        case .CharacteristicNotFound:
            return NSLocalizedString("Bluetooth characteristic not found", comment: "")
        case .PeripheralNotFound:
            return NSLocalizedString("Bluetooth peripheral not found", comment: "")
        case .Generic:
            return NSLocalizedString("Bluetooth generic error", comment: "")
        }
    }
}
